plugins/woocommerce/src/Internal/Shipping/ShippingService.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Shipping;
+
+use WC_Shipping_Zones;
+use WC_Shipping_Zone;
+use WP_Error;
+use WP_Http;
+use WC_Cache_Helper;
+
+/**
+ * A service class to manage shipping zones and their methods & locations.
+ *
+ * @internal
+ */
+class ShippingService {
+
+	/**
+	 * Get all shipping zones sorted by order.
+	 *
+	 * @return array Array of shipping zones sorted by zone_order.
+	 */
+	public function get_sorted_shipping_zones() {
+		// Get all zones including "Rest of the World".
+		$zones             = WC_Shipping_Zones::get_zones();
+		$rest_of_the_world = WC_Shipping_Zones::get_zone_by( 'zone_id', 0 );
+
+		// Add "Rest of the World" zone at the end with same structure as get_zones().
+		$rest_data                            = $rest_of_the_world->get_data();
+		$rest_data['zone_id']                 = $rest_of_the_world->get_id();
+		$rest_data['formatted_zone_location'] = array();
+		$rest_data['shipping_methods']        = $rest_of_the_world->get_shipping_methods( false, 'admin' );
+		$zones[0]                             = $rest_data;
+
+		// Sort zones by order.
+		uasort(
+			$zones,
+			function ( $a, $b ) {
+				return $a['zone_order'] <=> $b['zone_order'];
+			}
+		);
+
+		return $zones;
+	}
+
+	/**
+	 * Create a shipping zone from REST API request.
+	 *
+	 * @param array $params Request parameters.
+	 * @return WC_Shipping_Zone|WP_Error True on success, WP_Error on failure.
+	 */
+	public function create_shipping_zone( $params ) {
+		$zone   = new WC_Shipping_Zone( null );
+		$result = $this->update_shipping_zone( $zone, $params );
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+		return $zone;
+	}
+
+	/**
+	 * Update zone details from REST API request.
+	 *
+	 * @param WC_Shipping_Zone $zone zone to be updated.
+	 * @param array            $params Request parameters.
+	 * @return true|WP_Error True on success, WP_Error on failure.
+	 */
+	public function update_shipping_zone( $zone, $params ) {
```

Comment:
How would you feel about using `wp_parse_args()` to normalize the input and conveniently document all of the available parameters?

Reply:
Thanks for the pointer, I have added it to the update functions.